quinn-proto/src/congestion.rs

```diff
@@ -82,3 +91,14 @@ pub trait ControllerFactory {
 }
 
 const BASE_DATAGRAM_SIZE: u64 = 1200;
+
+/// Common congestion controller metrics
+#[non_exhaustive]
+pub struct ControllerMetrics {
```

Comment:
This needs `#[derive(Default)]` or `#[non_exhaustive]` will make it impossible to construct externally.

Reply:
Also, let's move this above the `ControllerFactory` definition.